client/src/Page.tsx

```diff
@@ -0,0 +1,210 @@
+import React, { useEffect, useState } from 'react'
+import { BaseText } from './components/Text'
+import { apis, buildClient } from './api'
+
+import { NotionRenderer } from 'react-notion-x'
+import 'react-notion-x/src/styles.css'
+
+import { Code } from 'react-notion-x/build/third-party/code'
+import TweetEmbed from 'react-tweet-embed'
+import { Collection } from 'react-notion-x/build/third-party/collection'
+import { Equation } from 'react-notion-x/build/third-party/equation'
+import { Modal } from 'react-notion-x/build/third-party/modal'
+import { Pdf } from 'react-notion-x/build/third-party/pdf'
+import {
+  extractTitle,
+  extractDescription,
+  extractPageCover,
+  extractPageEmoji,
+  makeEmojiDataUrl,
+  extractEmoji,
+  isValidNotionPageId,
+  parsePath,
+  urlNormalize
+} from '../../common/notion-utils'
+import { type ExtendedRecordMap } from 'notion-types'
+import { getPath, getSld, getSubdomain, titleEmbeddedMapPageUrl } from './utils'
+import { useTryCatch } from './hooks/useTryCatch'
+import { Navigate } from 'react-router-dom'
+import { BlankPage, LoadingScreen } from './components/Misc'
+import { LinkWrarpper } from './components/Controls'
+import { FlexColumn } from './components/Layout'
+import { Helmet } from 'react-helmet'
+import config from '../config'
+
+const Tweet = ({ id }: { id: string }): JSX.Element => {
+  return <TweetEmbed tweetId={id} />
+}
+
+const sld = getSld()
+const subdomain = getSubdomain()
+const pageIdOverride = parsePath(getPath().slice(1))
+
+const notion = config.embedPlatform === 'notion'
+
+const Page: React.FC = () => {
```

Comment:
use separate components for notion and substack - they have very different rendering logic

Reply:
Okay

Reply:
Done